hometasks/5-desigh-patterns/Shipment.ts

```diff
@@ -35,7 +40,17 @@ export class Shipment {
     }
 
     public getCost(): number {
-        return this.weight * 39;
+        switch (true) {
+            case this.weight <= PackageType.letter :
+
+                return this.weight * this.shippingStrategy.letterCharge;
+            case this.weight <= PackageType.package :
+
+                return this.weight * this.shippingStrategy.packageCharge;
+            default:
+
+                return this.shippingStrategy.getCost(this.weight);
+        }
```

Comment:
please consider of making this logic to be a part of strategy instead of creating fork of logic again in the Shipment.

```
abstract class Shipper {
  abstract readonly letterCharge: number;
  abstract readonly packageCharge: number;

  protected abstract getOversizedCost(weight: number): number;
  protected getPackageCost = (weight: number) => weight * this.packageCharge;
  protected getLetterCost = (weight: number) => weight * this.letterCharge;

  public getCost(weight: number) {
      if(weight > PackageType.package) {
         return this.getOversizedCost(weight);
      }

      return weight <= PackageType.letter 
             ? this.getLetterCost(weight) 
             : this.getPackageCost(weight) ;
  }
}
```
Then our strategies will have implementations for those parts that differ only
```
export class PacificParserShipper extends Shipper { 
   letterCharge = 0.51;
   packageCharge = 0.19;
   extraCharge =  0.02;
   
   getOversizedCost(weight: number) {
       return this.getPackageCost(weight) + weight * this.extraCharge;
   }
}

export class ChicagoSprintShipper extends Shipper {
   letterCharge = 0.42;
   packageCharge = 0.20;
   
   getOversizedCost(weight: number) {
       return this.getPackageCost(weight);
   }
}

export class AirEastShipper extends Shipper {
   letterCharge = 0.39;
   packageCharge = 0.25;
   extraCharge = 10;
   
   getOversizedCost(weight: number) {
       return this.getPackageCost(weight) + this.extraCharge;
   }
}
```
Then in client, there is only one line of code to be called
```
return this.shippingStrategy.getCost(this.weight);
```